### PR TITLE
[Doctests] Fix ignore bug and add more doc tests

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

This PR finishes all doctests for the speech models and fixes a bug with the newly added doctest flag.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/models/speech_to_text/modeling_speech_to_text.py [hunk 1]
```diff
@@ -780,7 +773,7 @@ def forward(
             attention_mask = self._get_feature_vector_attention_mask(inputs_embeds.shape[1], attention_mask)
             padding_mask = attention_mask.ne(1).long()
         else:
-            padding_mask = torch.zeros_like(inputs_embeds, dtype=torch.long)
+            padding_mask = torch.zeros(inputs_embeds.shape[:2], dtype=torch.long, device=inputs_embeds.device)
```
Comment (patrickvonplaten) [inline src/transformers/models/speech_to_text/modeling_speech_to_text.py:776]: There was a bug actually for which an additional test is added

File: conftest.py [hunk 2]
```diff
@@ -67,10 +67,12 @@ def pytest_sessionfinish(session, exitstatus):
 
 OutputChecker = doctest.OutputChecker
 
+
 class CustomOutputChecker(OutputChecker):
     def check_output(self, want, got, optionflags):
-        if IGNORE_RESULT and optionflags:
+        if IGNORE_RESULT & optionflags:
```
Comment (patrickvonplaten) [inline conftest.py:73]: @sgugger - I've dug a bit into this. `IGNORE_RESULTS` is actually not a bool but an integer so as it was coded now **all** doc tests were ignored. We need to use the bitwise and here I think (tested it locally and it works again now).

File: src/transformers/models/speech_to_text/modeling_speech_to_text.py [hunk 3]
```diff
@@ -1305,27 +1315,22 @@ def forward(
         >>> import torch
         >>> from transformers import Speech2TextProcessor, Speech2TextForConditionalGeneration
         >>> from datasets import load_dataset
-        >>> import soundfile as sf
 
         >>> model = Speech2TextForConditionalGeneration.from_pretrained("facebook/s2t-small-librispeech-asr")
         >>> processor = Speech2TextProcessor.from_pretrained("facebook/s2t-small-librispeech-asr")
 
 
-        >>> def map_to_array(batch):
```
Comment (patrickvonplaten) [inline src/transformers/models/speech_to_text/modeling_speech_to_text.py:1314]: old API should be removed

File: src/transformers/models/speech_to_text_2/modeling_speech_to_text_2.py [hunk 4]
```diff
@@ -35,13 +35,12 @@
 logger = logging.get_logger(__name__)
 
 _CONFIG_FOR_DOC = "Speech2Text2Config"
-_TOKENIZER_FOR_DOC = "Speech2Text2Tokenizer"
-_CHECKPOINT_FOR_DOC = "facebook/s2t-small-librispeech-asr"
+_CHECKPOINT_FOR_DOC = "facebook/s2t-wav2vec2-large-en-de"
 
 
 SPEECH_TO_TEXT_2_PRETRAINED_MODEL_ARCHIVE_LIST = [
-    "facebook/s2t-small-librispeech-asr",
```
Comment (patrickvonplaten) [inline src/transformers/models/speech_to_text_2/modeling_speech_to_text_2.py:43]: this is not speech2text2 it's speech2text

File: conftest.py (hunk 2 above)
Comment (sgugger) [reply conftest.py:73]: Ah, thanks for checking and fixing! I had tested it locally but I must have messed something up.

**Review by sgugger — APPROVED** [review]
Thanks for fixing!

File: src/transformers/models/speech_to_text/modeling_speech_to_text.py [hunk 5]
```diff
@@ -1167,6 +1155,28 @@ def forward(
         output_hidden_states=None,
         return_dict=None,
     ):
+        r"""
+        Returns:
+
+        Example:
+
+        ```python
```
Comment (sgugger) [inline src/transformers/models/speech_to_text/modeling_speech_to_text.py:1163]:
```suggestion
         ```python
```
This should fix the doc building error.

Comment (patrickvonplaten) [reply src/transformers/models/speech_to_text/modeling_speech_to_text.py:1163]: Thanks!

**Pull request merged**